### PR TITLE
Make available offline for ReceivedShare

### PR DESCRIPTION
**Pull request by hoangdat**
@dab246 please review the data and domain layer for `makeAvailableOffline`

https://user-images.githubusercontent.com/6462404/139422201-aeae1e4e-fe8b-42c4-a84d-ad48dfec3e85.mp4

### CODE REVIEW

File: data/lib/src/local/received_share_database_manager.dart [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+class ReceivedShareDatabaseManager extends LinShareDatabaseManager<ReceivedShare> {
+  final DatabaseClient _databaseClient;
+
+  ReceivedShareDatabaseManager(this._databaseClient);
+
+  @override
+  Future<bool> deleteAllData() async {
+    final receivedShareCache = await getListData();
+    var failedFileCount = 0;
+    receivedShareCache.forEach((receivedShare) async {
```
Comment (dab246) [inline data/lib/src/local/received_share_database_manager.dart:50]: Why haven't you fixed this yet?

File: domain/lib/src/usecases/received/get_all_received_shares_interactor.dart [hunk 2]
```diff
@@ -44,8 +46,23 @@ class GetAllReceivedSharesInteractor {
...
     try {
-      final receivedShares = await _receivedShareRepository.getAllReceivedShares();
-      return Right<Failure, Success>(GetAllReceivedShareSuccess(receivedShares));
+      final receivedShares = await _receivedShareRepository.getAllReceivedShares()
+        .onError((error, stackTrace) => _receivedShareRepository.getAllReceivedShareOffline());
+      final combinedReceivedShares = List<ReceivedShare>.empty(growable: true);
+
+      if (receivedShares.isNotEmpty) {
+        for (final received in receivedShares) {
+          final localReceivedShare = await _receivedShareRepository
```
Comment (dab246) [inline domain/lib/src/usecases/received/get_all_received_shares_interactor.dart:55]: Why don't you use `stream` here and do `combine` it in the `repository`?

File: data/lib/src/local/received_share_database_manager.dart (hunk 1 above)
Comment (hoangdat) [reply data/lib/src/local/received_share_database_manager.dart:50]: srr, to be update

File: domain/lib/src/usecases/received/get_all_received_shares_interactor.dart (hunk 2 above)
Comment (hoangdat) [reply domain/lib/src/usecases/received/get_all_received_shares_interactor.dart:55]: can you more details?

**Review by dab246 — APPROVED** [review]

**Pull request merged**